app/jobs/conversations/detect_language_job.rb

```diff
@@ -0,0 +1,38 @@
+class Conversations::DetectLanguageJob < ApplicationJob
+  queue_as :default
+
+  def perform(message_id)
+    @message = Message.find(message_id)
+
+    return if hook.blank?
+
+    text = @message.content[0...1500]
+
+    response = client.detect_language(
+      content: text,
+      parent: "projects/#{hook.settings['project_id']}"
+    )
+    update_conversation(response)
+  end
+
+  private
+
+  def update_conversation(response)
+    conversation = @message.conversation
+    current_language = response.languages.first.language_code
+    additional_attributes = conversation.additional_attributes.merge({ current_language: current_language })
+    # rubocop:disable Rails/SkipsModelValidations
+    conversation.update_columns(additional_attributes: additional_attributes)
```

Comment:
> The automation rule listener is tied to conversation_updated?

No. It is available for all the events

> if we use update_column, how will the subsequent automation rule be triggered?

We want to avoid triggering automation while updating the language. As this is happening as a background job to update additional_attribute for conversation_language when we create the message.